astropy/coordinates/tests/test_masked.py

```diff
@@ -0,0 +1,364 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+
+import numpy as np
+import pytest
+from numpy.testing import assert_array_equal
+
+import astropy.coordinates.representation as r
+import astropy.units as u
+from astropy.coordinates import FK5, SkyCoord
+from astropy.coordinates.matrix_utilities import rotation_matrix
+from astropy.utils.masked import Masked
+
+
+class MaskedSphericalSetup:
+    @classmethod
+    def setup_class(self):
```

Comment:
In principle, yes, but we've used `self` often in the tests and it oddly reads better when one doesn't assign `cls.XXX`. But happy to change too; it will make the distinction with `setup_method()` clearer.

Reply:
It is used all through astropy, wrongly or not. I don't think I should deviate from it here.

Reply:
If we don't fix it here, we should fix it Astropy-wide.

Reply:
> It is used all through astropy, wrongly or not.

note that despite syntacticly colored, `self` and `cls` are not keywords in Python, merely conventions. So, if one uses `self` as the name of the first argument in a `@classmethod`, the `self` variable is *still* binded to the *class* (indeed, there's no way to access any specific instance from this argument alone), so, I'm really not sure what you mean by "wrongly or not", because I don't see how it could possibly be correct (and is certainly very confusing). I agree that we should fix it at the package level. Apparently this doesn't exist as a linting rule already so I'll start by opening a feature request on ruff and see if we can get detection (maybe even autofixing) for free.

Reply:
Yes, let's auto-fix in a separate PR. Thanks!

Reply:
Another option would be to rewrite the tests to make use of module-level constants or `pytest` fixtures.

Reply:
Since `ruff` can autofix this, I just went ahead and did it. I do find `setup_class` considerably clearer than fixtures... Sorry for gruffness...